tools/testing/selftests/bpf/progs/read_vsyscall.c

```diff
@@ -1,16 +1,22 @@
 // SPDX-License-Identifier: GPL-2.0
 /* Copyright (C) 2024. Huawei Technologies Co., Ltd */
+#include "vmlinux.h"
 #include <linux/types.h>
 #include <bpf/bpf_helpers.h>
 
 #include "bpf_misc.h"
 
 int target_pid = 0;
 void *user_ptr = 0;
-int read_ret[8];
+int read_ret[9];
 
 char _license[] SEC("license") = "GPL";
 
+/*
+ * This is the only kfunc, the others are helpers
+ */
+int bpf_copy_from_user_str(void *dst, u32, const void *) __ksym;
```

Comment:
`__weak __ksym`, also use the exact same types for arguments, as depending on your setup this declaration will also be coming from vmlinux.h, and so it has to be at least compatible (or best is to check what do we get in vmlinux.h)

Reply:
It doesn't seem to like if I include additional decl tags like `__user`. Also this new function wasn't in my vimlinux.h when I generated it from bpftool. Is it named something different?

Reply:
it's fine, just make sure to have `__weak`